app/models/category.rb

```diff
@@ -3,6 +3,16 @@ class Category < ActiveRecord::Base
 
   belongs_to :data_list
 
+  def store_cat
+    self.order = 0
+    # Even if parent_id is nil this does the right thing.
+    last_child = Category.where(ancestry: parent_id).order('categories.order desc').limit(1)[0]
+    if last_child && last_child.order
```

Comment:
Just kidding. I was thinking of data_list_measurements. but we really should change this column to something like `list_order` as well.